src/control/server/ctl_system.go

```diff
@@ -105,12 +165,37 @@ func (svc *ControlService) SystemStop(ctx context.Context, req *ctlpb.SystemStop
 		return nil, err
 	}
 
-	svc.log.Debug("received SystemStop RPC; proceeding to shutdown DAOS system")
+	svc.log.Debug("Received SystemStop RPC")
+
+	// TODO: consider locking to prevent join attempts when shutting down
```

Comment:
IMO it'd be best to just do it now; that way it's not forgotten and we find any bugs caused by this sooner rather than later.

Reply:
I put a Lock in but it stopped the prep_shutdown from propagating, that's why the to do is there